src/Pages/MembershipApplication/MembershipForm.js

```diff
@@ -38,50 +72,65 @@ export default function MembershipForm(props) {
       addon: !usernameAvailable && (
         <p className='unavailable'>User already exists!</p>
       ),
-      handleChange: e => setEmail(e.target.value)
+      ifRequirementsNotMet: checkEmailInput(),
+      handleChange: e => setEmail(e.target.value),
     },
+
     {
       label: 'Password (8 or more characters)',
       type: 'password',
       addon: !passwordValid && (
         <p className='unavailable'>
-          Password requires one uppercase character and one number.
+          Password requires one uppercase character, one number
+          and at least 8 characters
         </p>
       ),
-      handleChange: e => setPassword(e.target.value)
+      ifRequirementsNotMet: password.length<8 && clickSubmitted && (
```

Comment:
I don't think there's any point in checking password length, since you aren't checking the other requirements. Just using `passwordValid` might be good enough.